### PR TITLE
Fixes compiler warnings

### PR DESCRIPTION
**Pull request by t-young31**
Fixes the compiler warnings (1.ii. in #41 ). Unfortunately GCC throws a lot of `-Wmaybe-uninitialised` warnings that don't seem to be easily solvable so these are ignored.

- Compile on GitHub actions with `-Werror`
- Changes from printing error to throwing exceptions
- Alters the number of output matrices to be consistent with `openandorder.cpp`
- Initialises a few potentially uninitialised variables

***

### Dependencies

- #51 
- #52  : This PR assumes they can be deleted

### CODE REVIEW

File: tdms/src/iterator.cpp [hunk 1]
```diff
@@ -743,7 +727,8 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
     if( num_fields==6 ){
       for(int i=0;i<num_fields;i++){
 	element = mxGetField( (mxArray *)prhs[input_counter], 0, C_elements[i]);
-      
+  string element_name = C_elements[i];
```
Comment (giordano) [inline tdms/src/iterator.cpp:730]: Indentation is off

File: tdms/src/iterator.cpp [hunk 2]
```diff
@@ -778,17 +763,17 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
...
 	}
 	else
-	  mexfprintf("Incorrect dimension on C");
+	  throw runtime_error("Incorrect dimension on C");
       }
     }
     else{
       for(int i=0;i<num_fields;i++){
 	element = mxGetField( (mxArray *)prhs[input_counter], 0, C_elements_disp_ml[i]);
+  string element_name = C_elements_disp_ml[i];
```
Comment (giordano) [inline tdms/src/iterator.cpp:776]: Indentation

File: tdms/src/iterator.cpp [hunk 3]
```diff
@@ -881,19 +862,17 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
 	  Dbz = mxGetPr(element);
 	}
 	else{
-	  sprintf(message_buffer, "element D.%s not handled",D_elements[i]);
-	  mexfprintf(message_buffer);
+	  throw runtime_error("element D."+element_name+" not handled");
 	}
       }
       else
```
Comment (giordano) [inline tdms/src/iterator.cpp:868]: Add brackets

File: tdms/src/iterator.cpp [hunk 4]
```diff
@@ -936,18 +913,16 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
...
 	}
 	else{
-	  sprintf(message_buffer, "element freespace.%s not handled",freespace_elements[i]);
-	  mexfprintf(message_buffer);
+	  throw runtime_error("element freespace. "+element_name+" not handled");
 	}
       }
       else
-	mexfprintf("Incorrect dimension on freespace");
+	throw runtime_error("Incorrect dimension on freespace");
```
Comment (giordano) [inline tdms/src/iterator.cpp:920]: Indentation and brackets

File: tdms/src/iterator.cpp [hunk 5]
```diff
@@ -1070,9 +1036,15 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
 
 	//dimptr = (int *)mxGetDimensions(element);
 	dimptr_out = mxGetDimensions(element);
+  string field_name = interface_fields[i];
```
Comment (giordano) [inline tdms/src/iterator.cpp:1039]: Indentation

File: tdms/src/iterator.cpp [hunk 6]
```diff
@@ -1260,13 +1231,11 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
...
 	else{
-	  sprintf(message_buffer, "element grid_labels.%s not handled",grid_labels_fields[i]);
-	  mexfprintf(message_buffer);
+	  throw runtime_error("element grid_labels. "+grid_label+" not handled");
 	}
       }
       else{
-	sprintf(message_buffer, "Incorrect dimension on grid_labels.%s",Cmaterial_elements[i]);
-	mexfprintf(message_buffer);
+	throw runtime_error("Incorrect dimension on grid_labels. " + material_element);
```
Comment (giordano) [inline tdms/src/iterator.cpp:1238]: Indentantion

File: tdms/src/iterator.cpp [hunk 7]
```diff
@@ -2028,45 +1961,37 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
...
+          dimptr_out = mxGetDimensions(element);
+          fprintf(stderr,"found vertices (%d x %d)\n", dimptr_out[0], dimptr_out[1]);
+          vertices = castMatlab2DArrayInt((int *)mxGetPr( (mxArray *)element), dimptr_out[0], dimptr_out[1]);
+          //fprintf(stderr,"vertices[1000] = %d %d %d\n",vertices[0][10],vertices[1][10],vertices[2][10]);
+          nvertices =  dimptr_out[0];
+          //convert vertices to index base 0
+
+          for( int j=0;j<nvertices;j++)
+            for( int k=0;k<3;k++)
+              vertices[k][j] = vertices[k][j] - 1;
```
Comment (giordano) [inline tdms/src/iterator.cpp:1979]: Brackets

File: tdms/src/iterator.cpp [hunk 8]
```diff
@@ -2028,45 +1961,37 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
...
-	    ncomponents = dimptr_out[1];
-	}
-	fprintf(stderr,"found components (%d)\n",ncomponents);
+        if( !mxIsEmpty(element) ){
+          dimptr_out = mxGetDimensions(element);
+          components = (int *)mxGetPr( (mxArray *)element);
+          if( dimptr_out[0] > dimptr_out[1] )
+            ncomponents = dimptr_out[0];
+          else
+            ncomponents = dimptr_out[1];
```
Comment (giordano) [inline tdms/src/iterator.cpp:1990]: Brackets

File: tdms/src/iterator.cpp [hunk 9]
```diff
@@ -3392,6 +3308,7 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
 		}
 
 
+      Enp1 = 0.0;
```
Comment (giordano) [inline tdms/src/iterator.cpp:3311]: Indentantion

File: tdms/src/iterator.cpp [hunk 10]
```diff
@@ -6408,12 +6325,9 @@ if(runmode == rm_complete && (nvertices>0) )
     freeCastMatlab3DArray(KsourceR,((int)(J1[0]-J0[0]+1.)));
   }
 
-if( !((N_fieldsample_i == 0) || (N_fieldsample_j == 0) || (N_fieldsample_k == 0) || (N_fieldsample_n == 0)) ){
-  freeCastMatlab4DArray( fieldsample, N_fieldsample_k,N_fieldsample_n);
-  freeCastMatlab4DArray( fieldsample_x, N_fieldsample_k,N_fieldsample_n);
-  freeCastMatlab4DArray( fieldsample_y, N_fieldsample_k,N_fieldsample_n);
-  freeCastMatlab4DArray( fieldsample_z, N_fieldsample_k,N_fieldsample_n);
```
Comment (giordano) [inline tdms/src/iterator.cpp:6415]: Why were these removed?

**Review by giordano — APPROVED** [review]
Looks good to me, mainly stylistic comments

File: tdms/src/iterator.cpp (hunk 1 above)
Comment (t-young31) [reply tdms/src/iterator.cpp:730]: trying to minimise the diff but will change 👍🏼

File: tdms/src/iterator.cpp (hunk 10 above)
Comment (t-young31) [reply tdms/src/iterator.cpp:6415]: `fieldsample_x` etc. are now not allocated (assuming #52 means they can be deleted)

**Review by prmunro — APPROVED** [review]
